x-pack/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -544,11 +546,147 @@ export function jobRoutes({ router, mlLicense }: RouteInitialization) {
     },
     mlLicense.fullLicenseAPIGuard(async (context, request, response) => {
       try {
-        const options = {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.categories', {
           jobId: request.params.jobId,
           categoryId: request.params.categoryId,
-        };
-        const results = await context.ml!.mlClient.callAsCurrentUser('ml.categories', options);
+        });
+        return response.ok({
+          body: results,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId/model_snapshots Get model snapshots by job ID
+   * @apiName GetModelSnapshots
+   * @apiDescription Returns the model snapshots for the specified job ID
+   *
+   * @apiSchema (params) getModelSnapshotsSchema
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/model_snapshots',
+      validate: {
+        params: getModelSnapshotsSchema,
+      },
+      options: {
+        tags: ['access:ml:canGetJobs'],
+      },
+    },
+    mlLicense.fullLicenseAPIGuard(async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.modelSnapshots', {
+          jobId: request.params.jobId,
+        });
+        return response.ok({
+          body: results,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId/model_snapshots/:snapshotId Get model snapshots by job ID and snapshot ID
+   * @apiName GetModelSnapshotsById
+   * @apiDescription Returns the model snapshots for the specified job ID and snapshot ID
+   *
+   * @apiSchema (params) getModelSnapshotsSchema
+   */
+  router.get(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/model_snapshots/{snapshotId}',
+      validate: {
+        params: getModelSnapshotsSchema,
+      },
+      options: {
+        tags: ['access:ml:canGetJobs'],
+      },
+    },
+    mlLicense.fullLicenseAPIGuard(async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.modelSnapshots', {
+          jobId: request.params.jobId,
+          snapshotId: request.params.snapshotId,
+        });
+        return response.ok({
+          body: results,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/:jobId/model_snapshots/:snapshotId/_update update model snapshot by snapshot ID
+   * @apiName UpdateModelSnapshotsById
+   * @apiDescription Updates the model snapshot for the specified snapshot ID
+   *
+   * @apiSchema (params) getModelSnapshotsSchema
```

Comment:
could you please add ` @apiSchema (body) updateModelSnapshotSchema `

Reply:
Changed in ac22c8ce86162fd9d4db4687850de4ae33873f4b